### PR TITLE
[EWL-5301] SG2 | Create "Subscribe Promo" Molecule (Cat/Subcat)

### PR DESCRIPTION
**Pull request by csj1012**
## Ticket(s)

**Jira Ticket**
- [EWL-5301: SG2 | Create "Subscribe Promo" Molecule (Cat/Subcat)](https://issues.ama-assn.org/browse/EWL-5301)

## Description
Adds the "Subscribe Promo" Molecule.

## To Test
- [x] `gulp serve`
- [x] Given that I am viewing SG2
- [x] When I click "Molecules" in the menu
- [x] Then I should see "Subscribe Promo"
- [x] When I click "Subscribe Promo"
- [x] Then I should see the purple box with white text
- [x] And I should optionally see the subject title white all-caps at the top 
- [x] And I should see the the Title in white
- [x] And I should see the subtext in small white text
- [x] And I should see an email input
- [x] And I should see a white "Subscribe" button with purple text
- [x] Did you test in IE 11?

## Visual Regressions

- [x] **Before proceeding:** Run visual regression tests locally to ensure new work does not introduce new bugs.

- [x] _If the new work does your work introduce visual regressions into existing work in the Style Guide (or in a Drupal 8 environment)_, either call these out here or address them before marking the PR as `ready for review`.

- [x] If you are creating or updating a pattern be sure you have created or updated the [scenario in `backstop.json`](ama-style-guide-2/styleguide/backstop.json) and have included a new or updated screenshot in [`bitmaps_reference`](ama-style-guide-2/styleguide/backstop_data/bitmaps_reference).

## Relevant Screenshots/GIFs
screen shot 2018-06-13 at 5 05 18 pm

## Remaining Tasks
- [ ] Updating [The Glossary](https://issues.ama-assn.org:8446/confluence/display/DTD/Glossary?src=contextnavpagetreemode) 

## Additional Notes
N/A

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

**Review by labbydev — CHANGES_REQUESTED** [review]
Looking at the Zeplin, the "Subscribe" button on the promo should be centered in the container. Also the hover makes the button seemingly disappear. This is something that we should address with UX likely. 

**SG2 Default** 
screen shot 2018-06-14 at 11 37 15 am

**SG2 Hover**
screen shot 2018-06-14 at 11 37 19 am

----

**Zeplin**
screen shot 2018-06-14 at 11 40 43 am

----
I'm also seeing about 41 backstop failures including:
* atoms -> This isn't failing but it's not testing all of them since the page doesn't scroll correctly. FWIW I don't think we should be doing all atoms together and this test doesn't work anyway since the page doesn't scroll. _If_ we are going to do all atoms together then we need to be using the "open in new window" URL.
* event stub -> change in bottom padding
* hub hero as fifty fifty video -> This is a change in the YouTube preview which I'm willing to look past
* info card as location -> This is a change in the google map which I'm willing to look past.
* search header
* masthead -> in some cases this is a bug in the reference image
* event list -> cascade of changes in event stub molecule
* footer -> change in padding under social media icons
* hub heros -> cascade of hub hero as fifty fifty video
* resource -> change in YouTube screenshot
* event listing -> cascade of changes in event stub molecule

----

Regression in colors

When going to the colors atom in base then I should see swatches of the colors and that functionality is no longer present

screen shot 2018-06-14 at 11 55 40 am

**Comment by csj1012** [comment]
Changes (from UX): 
height of button: 40px / 60px (mobile) 
center button

**Comment by csj1012** [comment]
@labbydev I've adjusted the button styling. Recommendation to address the backstop issues and visual regressions separately as they seem to be present in `develop` and outside the scope of this ticket. This is ready for re-review.

**Review by JaeBea — CHANGES_REQUESTED** [review]
Hi @chelsiejohnston,

I have one update needed to merge and a couple of questions on your implementation.

1. When running backstop I am getting errors for the subscribe promo molecule.
image

Implementation Questions
1. It is my understanding that we should reuse atoms and molecules as much as possible in this style guide. Did UX provide additional information about the html structure that prevents the use of the heading and paragraph atoms?

2. Also some of the added styles seems to repeat the ama__h4 and ama__h2 styles which I believe can be applied to any element. Can those styles be reused here?

**Comment by csj1012** [comment]
Hi @JaeBea - the backstop diff is an intentional change as the button is meant to be centered. I've updated the reference images, sorry about that.

Your understanding about reusing SG patterns is correct. The reason I didn't originally use heading elements was that the JIRA ticket didn't specifically call for them, and that semantically I don't love them being `<h2>`s and `<h4>` because it doesn't really make sense. That said, in doing this I broke the pattern we usually use for the promos (which is to just use the headings anyway). So, I went ahead and just changed them all to the expected pattern includes (which removes the need to extend `ama__h2` and `ama__h4` styles in the named CSS rules).

So anyway, this is ready for re-review! :)

File: styleguide/source/assets/scss/02-molecules/_subscribe-promo.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      @include type($myriad-pro, $font-weight-regular);
+      font-style: italic;
+    }
+
+    label {
+      visibility: hidden;
+    }
+  }
+
+  button {
```
Comment (designerneil) [inline styleguide/source/assets/scss/02-molecules/_subscribe-promo.scss:24]: Shouldn't this be using the button.scss styling?

Comment (designerneil) [reply styleguide/source/assets/scss/02-molecules/_subscribe-promo.scss:24]: If it's a custom button should we add to the button mixins so we can reuse it?

Comment (csj1012) [reply styleguide/source/assets/scss/02-molecules/_subscribe-promo.scss:24]: This styling does already use the button.scss styling but is also different and so needs to override several of the rules there. I don't know if we'll need to reuse it so I'd rather not change this for now. @labbydev @JaeBea do you have any thoughts about this?

Comment (labbydev) [reply styleguide/source/assets/scss/02-molecules/_subscribe-promo.scss:24]: This seems like a rather niche case, so I'm fine with not adding a button variant for it.

**Review by labbydev — CHANGES_REQUESTED** [review]
It looks like you forgot to update the hover background color. Other than that everything looks good to me. 

screen shot 2018-06-21 at 11 47 38 am

**Comment by csj1012** [comment]
All backstop tests are now passing on this branch. I approved and committed one failing test for "visual subcategory exploration with images" that did not appear to have any visual changes inherent. This PR is ready for review.

**Review by labbydev — APPROVED** [review]

**Pull request merged**